packages/react/src/auto/hooks/useJSONInputController.tsx

```diff
@@ -1,48 +1,54 @@
 import { useCallback, useEffect, useState } from "react";
-import { type Control } from "react-hook-form";
+import deepEqual from "react-fast-compare";
+import type { UseControllerProps } from "react-hook-form";
+import { isFailedJSONParse, type FailedJSONParse } from "../../validationSchema.js";
 import { useStringInputController } from "./useStringInputController.js";
 
-export const useJSONInputController = (props: {
-  field: string; // The field API identifier
-  control?: Control<any>;
-}) => {
-  const { field, control } = props;
-  const stringInputController = useStringInputController({ field, control });
+export const useJSONInputController = (
+  props: {
+    field: string;
+  } & Omit<UseControllerProps, "name">
+) => {
+  const stringInputController = useStringInputController(props);
+  const jsonValue = stringInputController.value;
+  const isParseError = isFailedJSONParse(jsonValue);
 
-  const [error, setError] = useState<Error | undefined>();
-  const [jsonValue, setJSONValue] = useState<any>(stringInputController.value);
-  const [stringValue, setStringValue] = useState<string>(JSON.stringify(jsonValue, null, 2));
+  const [string, setString] = useState<string>(stringInputController.value && !isParseError ? JSON.stringify(jsonValue, null, 2) : "");
 
-  const onStringValueChange = useCallback(
+  // propagate imperative changes from the form state into the string input, but only if the parsed representations aren't equal
+  useEffect(() => {
+    if (typeof jsonValue != "undefined" && !isParseError) {
+      try {
+        const stringJSONValue = JSON.parse(string);
```

Comment:
`JSON.parse("")` throws an error if the string is empty. This makes it impossible to clear out an existing record's JSON field value

Reply:
Nice, fixed!